### PR TITLE
Merged config document for account with system defaults.

### PR DESCRIPTION
**Pull request by jamhed**
merged config for account (account/reseller/system/system_default)

### CODE REVIEW

**Comment by jamhed** [comment]
@fenollp, unit tests?

**Comment by fenollp** [comment]
yep unit tests for kz_json & kz_json_schema (please)

**Comment by jamhed** [comment]
@fenollp, I've added tests for flatten/default_object schema functions, there is no need to add more tests to kz_json as they are there already.

File: core/kazoo_json/test/kz_json_schema_test.erl [hunk 1]
```diff
@@ -145,3 +145,27 @@ validate_v4_test_() ->
...
+
+flatten_test() ->
+    Schema = kz_json:decode(<<"{\"properties\":{\"caller_id\":{\"description\":\"The default caller ID parameters\",\"type\":\"object\",\"properties\":{ \"internal\":{\"description\":\"The default caller ID used when dialing internal extensions\",\"type\":\"object\",\"properties\":{\"name\":{\"name\":\"Called ID Internal Name\",\"description\":\"The caller id name for the object type\",\"type\":\"string\",\"maxLength\":30}}},\"external\":{\"description\":\"The default caller ID used when dialing external numbers\",\"type\":\"object\",\"properties\":{\"name\":{\"name\":\"Caller ID External Name\",\"description\":\"The caller id name for the object type\",\"type\":\"string\",\"maxLength\":15}}},\"emergency\":{\"description\":\"The caller ID used when external, internal, or emergency is not defined\",\"type\":\"object\",\"properties\":{\"name\":{\"name\":\"Caller ID Emergency Name\",\"description\":\"The caller id name for the object type\",\"type\":\"string\",\"maxLength\":15,\"default\":\"emer_default\"}}}},\"default\":{}}}}">>),
+    Flat = kz_json_schema:flatten(Schema),
+    [?_assertMatch(Flat, {[
```
Comment (fenollp) [inline core/kazoo_json/test/kz_json_schema_test.erl]: Shouldn't these be _assertEqual instead? (they probably end up being the same in this specific case, but intent?)

Comment (jamhed) [inline core/kazoo_json/test/kz_json_schema_test.erl]: first, you're commenting outdated commit, and second, i don't really understand your comment :)

File: core/kazoo_config/src/kapps_config_util.erl [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+-spec load_config_from_system(api_binary(), ne_binary()) -> {ok, kz_json:object()} | {error, any()}.
+load_config_from_system(_Account, Config) ->
+    kz_json:get_value(<<"default">>, ensure_value(kapps_config:get_category(Config))).
+
+-spec load_default_config(api_binary(), ne_binary()) -> {ok, kz_json:object()}.
+load_default_config(_Account, Config) ->
+    Schema = system_schema(Config),
+    {ok, kz_doc:set_id(kz_json_schema:default_object(Schema), doc_id(Config))}.
+
+ensure_value({ok, JObj}) -> JObj;
```
Comment (icehess) [inline core/kazoo_config/src/kapps_config_util.erl]: `esure_value/1` and `maybe_new/1` has same functionality, maybe merge it into one?

File: core/kazoo_json/test/kz_json_schema_test.erl [hunk 3]
```diff
@@ -145,3 +145,22 @@ validate_v4_test_() ->
...
+    Default = kz_json_schema:default_object(Schema),
+    [?_assertEqual({[{<<"caller_id">>,{[{<<"emergency">>,{[{<<"name">>,<<"emer_default">>}]}}]}}]}, Default)].
+
+flatten_sms_schema_test() ->
+    Flat = kz_json_schema:flatten(get_schema_sms()),
+    [?_assertEqual(Flat, {[
+        {[<<"outbound">>,<<"options">>,<<"default">>], {[{<<"delivery_mode">>,2},{<<"mandatory">>,true}]}},
+        {[<<"outbound">>,<<"options">>,<<"description">>], <<"sms options">>},
+        {[<<"outbound">>,<<"options">>,<<"type">>],<<"object">>}]
+    })].
```
Comment (k-anderson) [inline core/kazoo_json/test/kz_json_schema_test.erl]: diff --git a/core/kazoo_json/test/kz_json_schema_test.erl b/core/kazoo_json/test/kz_json_schema_test.erl
index 14d1212a1..b0f11a484 100644
--- a/core/kazoo_json/test/kz_json_schema_test.erl
+++ b/core/kazoo_json/test/kz_json_schema_test.erl
@@ -160,7 +160,7 @@ default_object_test() ->
 flatten_sms_schema_test() ->
     Flat = kz_json_schema:flatten(get_schema_sms()),
     [?_assertEqual(Flat, {[
-        {[<<"outbound">>,<<"options">>,<<"default">>], {[{<<"delivery_mode">>,2},{<<"mandatory">>,true}]}},
-        {[<<"outbound">>,<<"options">>,<<"description">>], <<"sms options">>},
-        {[<<"outbound">>,<<"options">>,<<"type">>],<<"object">>}]
-    })].
+                           {[<<"outbound">>,<<"options">>,<<"default">>], {[{<<"delivery_mode">>,2},{<<"mandatory">>,true}]}},
+                           {[<<"outbound">>,<<"options">>,<<"description">>], <<"sms options">>},
+                           {[<<"outbound">>,<<"options">>,<<"type">>],<<"object">>}]
+                         })].

**Pull request merged**